swagger.yaml

```diff
@@ -75,6 +77,78 @@ paths:
             text/plain:
               schema:
                 type: string
+  /projects/{id}/pdf:
+    post:
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: integer
+          required: true
+          description: Numeric ID of the affected project
+      tags:
+        - compile
+      summary: Returns the compiled PDF for a given project and compiles it if needed.
+      description: |-
+        Returns a pdf, which will be compiled if needed. Takes in options to control the compilation process.
+      operationId: compile
+      requestBody:
+        description: Compile options
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CompileOptions'
+      responses:
+        200:
+          description: PDF file
+          content: 
+            application/pdf:
+              schema:
+                type: string
+                format: binary
+        400:
+          description: Malformed request body
+        401:
+          description: Unauthorized to execute operation
+        415:
+          description: Unsupported media type
+        422:
+          description: Request validation errors (e.g. missing required fields in options)
```

Comment:
I think the convention is that missing required fields in request body are 400. Spring works this way by default. But maybe this is fine too

Reply:
From what I could gather it seems that actually sending 422 is more common in this situation, but I might be wrong. The newest RFC also makes it sound like 422 is more in line with this situation, but it doesn't explicitly say. This is the default of axum and other frameworks like ruby on rails and flask

Reply:
alrighty then leave it